### DISCUSSION TITLE
browser.electron not found

### DISCUSSION THREAD

**Discussion by jgresham** [webdriverio/webdriverio#11856] [Technical Discussion]
Sorry if I missed this in the docs. I followed them but I'm getting the error that electron is not found when running. Do I need to tell Typescript to ignore the warning somehow?

<img width="842" alt="Captura de pantalla 2023-12-11 a la(s) 9 03 50 p m" src="https://github.com/webdriverio/webdriverio/assets/3721291/19779c93-b3ff-41b6-87e1-9c1da1e8134e">

**Comment by jgresham** [comment]
Will open on electron repo

**Comment by christian-bromann** [comment]
@jgresham you will have to make sure the types are loaded by adding `wdio-electron-service` to your `"types"` in the `tsconfig.json`. This is not related to Electron.

**Reply by jgresham** [reply]
Thanks!

<img width="579" alt="Captura de pantalla 2023-12-12 a la(s) 7 22 47 a m" src="https://github.com/webdriverio/webdriverio/assets/3721291/5bcdcb8f-2e18-499e-a75e-e6d475490e62">


Although, I'm having a similar issue with importing `'wdio-electron-service/main'` and `'wdio-electron-service/preload'` conditionally. I've tried many combinations of the following without any success: 
```
    "types": ["@wdio/globals/types", "wdio-electron-service", "wdio-electron-service/preload", "wdio-electron-service/main"],
    "typeRoots": ["./node_modules", "./node_modules/@types", "./@types"]
```

**Reply by christian-bromann** [reply]
Do you have `"moduleResolution": "node16",` set in your `tsconfig.json`?

**Reply by christian-bromann** [reply]
Also, try to setup a new Electron+WebdriverIO project using `npm init wdio@latest ./test` and compare that setup with yours.

**Reply by jgresham** [reply]
Ok using "moduleResolution": "node16" and the latest and `test` was being excluded in `tsconfig`.
Thank you for the help and work on this project! I'm really excited to find Webdriver and this plugin. It is exactly what I needed for my app!

**Reply by christian-bromann** [reply]
Awesome 🙌

**Comment by arjundangi01** [comment]
Please Help me 

I did above things and my ts error for  browser.electron is solved, but I can not access further like electron.app  electron.browserWindow

I followed this doc - https://www.electronjs.org/docs/latest/tutorial/automated-testing

and I tried it's first test 

import { browser, $, expect } from '@wdio/globals'

```
describe('when the make smaller button is clicked', () => {
  it('should decrease the window height and width by 10 pixels', async () => {
    const boundsBefore = await browser.electron.browserWindow('getBounds')
    expect(boundsBefore.width).toEqual(210)
    expect(boundsBefore.height).toEqual(310)

    await $('.make-smaller').click()
    const boundsAfter = await browser.electron.browserWindow('getBounds')
    expect(boundsAfter.width).toEqual(200)
    expect(boundsAfter.height).toEqual(300)
  })
})
```
